### PR TITLE
fix: don't throw on grid details cell focus

### LINKED ISSUE #2727: Clicking button in Details will cause JavaScript exception on console
Vaadin 23.0.0.rc1 and Vaadin 22.0.6

How to reproduce
- Open attached MasterDetailView
- Click a row to see details
- Click the button
- Check the error in the console

masterdetail:6 There seems to be an error in Vaadin Grid:
Unexpected value for section: details
Please submit an issue to https://github.com/vaadin/flow-components/issues/new/choose
console.error @ masterdetail:6
eval @ FlowBootstrap.js?f80c:67
_onContentFocusIn @ vaadin-grid-keyboard-navigation-mixin.js?709a:734

Code to reproduce

@PageTitle("Master-Detail")
@Route(value = "masterdetail")
public class MasterDetailView extends Div {

	private Grid<Vector<String>> grid = new Grid<>();

	public MasterDetailView() {
		setSizeFull();

		// Configure Grid
		grid.addColumn(item -> item.get(0)).setAutoWidth(true);
		grid.addColumn(item -> item.get(1)).setAutoWidth(true);

		grid.setItems(createData());
		grid.addThemeVariants(GridVariant.LUMO_NO_BORDER);
		grid.setHeightFull();

		grid.setItemDetailsRenderer(createPersonDetailsRenderer());

		add(grid);

	}

	private List<Vector<String>> createData() {
		Random rand = new Random();
		return IntStream.range(0, 20).mapToObj(i -> {
			Vector<String> vector = new Vector<>(2);
			vector.add("email" + i + "@domain.com");
			vector.add("" + rand.nextInt(20));
			return vector;
		}).collect(Collectors.toList());
	}

	private ComponentRenderer<PersonDetailsFormLayout, Vector<String>> createPersonDetailsRenderer() {
		return new ComponentRenderer<>(PersonDetailsFormLayout::new, PersonDetailsFormLayout::setPerson);
	}

	private class PersonDetailsFormLayout extends VerticalLayout {
		private final TextField emailField = new TextField("Email address");
		private final TextField phoneField = new TextField("Phone number");

		public PersonDetailsFormLayout() {
			Stream.of(emailField, phoneField).forEach(field -> {
				add(field);
			});

			Button btn = new Button("do it");
			btn.addClickListener(e -> {
				Notification.show("clicked");
			});
			add(btn);
		}

		public void setPerson(Vector<String> person) {
			emailField.setValue(person.get(0));
			phoneField.setValue(person.get(1));
		}
	}
}

### PR DESCRIPTION
**Pull request by tomivirkki**
Fixes #2727 

This PR makes the connector not throw when a `cell-focus` event originates from "details" section.

NOTE:
This is merely a bugfix to avoid the client-side exception, and doesn't add the missing support for details cell-focus events. Adding the missing feat is a separate topic and requires some design (for example, do we want to add "details" to the `GridSection` enum or use the existing "body" section with `null` column for details cell focus...)

### CODE REVIEW

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js [hunk 1]
```diff
@@ -1037,7 +1037,7 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
         const expectedSectionValues = ['header', 'body', 'footer'];
 
         if(expectedSectionValues.indexOf(eventContext.section) === -1) {
-          throw new Error('Unexpected value for section: ' + eventContext.section);
+          return;
```
Comment (sissbruecker) [inline vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js:1040]: Quick sanity check: Wouldn't this still throw an exception on the server-side? See here: https://github.com/vaadin/flow-components/blob/master/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/CellFocusEvent.java#L159-L161

Not sure if that is an issue or not.

**Review by sissbruecker — CHANGES_REQUESTED** [review]

File: vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridDetailsRowIT.java [hunk 2]
```diff
@@ -50,6 +50,14 @@ public void gridTwoItemsSelectedWhenOpen() {
         assertElementHasButton(detailsElement.get(1), "Person 2");
     }
 
+    @Test
+    public void shouldNotThrowOnDetailsClick() {
```
Comment (sissbruecker) [inline vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridDetailsRowIT.java:54]: Not sure how valuable this test is. This kind of implies that someone would add back an exception, or that some other random error might occur at some point.

Comment (tomivirkki) [reply vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridDetailsRowIT.java:54]: Agreed, it's kind of an odd thing to test. This applies to pretty much every test in the repo that uses `checkLogsForErrors()`.

Comment (sissbruecker) [reply vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridDetailsRowIT.java:54]: Alright, just wanted to get your opinion on this

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**